text/0000-rustdoc-json.md

```diff
@@ -0,0 +1,1124 @@
+`items`  | [[ID](#ID)] | The list of Items contained within this module. The order of definitions is preserved.
+
+### `kind == "function"`
+
+Name       | Type     | Description
+-----------|----------|----------------------------------------------------------------------------
+`decl`     | [FnDecl](#FnDecl) | Information about the function signature, or declaration.
+`generics` | [Generics](#Generics) | Information about the function's type parameters and `where` clauses.
+`header`   | String   | `"const"`, `"async"`, `"unsafe"`, or a space separated combination of those modifiers.
+`abi`      | String   | The ABI string on the function. Non-`extern` functions have a `"Rust"` ABI, whereas `extern` functions without an explicit ABI are `"C"`. See [the reference](https://doc.rust-lang.org/reference/items/external-blocks.html#abi) for more details.
+
+### `kind == "struct" || "union"`
+
+Name          | Type     | Description
+--------------|----------|-------------------------------------------------------------------------
+`struct_type` | String   | Either `"plain"` for braced structs, `"tuple"` for tuple structs, or `"unit"` for unit structs.
+`generics`    | [Generics](#Generics) | Information about the struct's type parameters and `where` clauses.
+`fields_stripped` | bool | Whether any fields have been removed from the result, due to being private or hidden.
+`fields`      | [[ID](#ID)] | The list of fields in the struct. All of the corresponding Items have `kind == "structfield"`.
+`impls`       | [[ID](#ID)] | All impls (both trait and inherent) for this type. All of the corresponding Items have `kind = "impl"`
+
+### `kind == "structfield"`
+
+Name          | Type     | Description
+--------------|----------|-------------------------------------------------------------------------
+`type`        | [Type](#Type) | The type of this field.
+
+### `kind == "enum"`
+
+Name          | Type     | Description
+--------------|----------|-------------------------------------------------------------------------
+`generics`    | [Generics](#Generics) | Information about the enum's type parameters and `where` clauses.
+`fields`      | [[ID](#ID)]     | The list of variants in the enum. All of the corresponding Items have `kind == "variant"`.
+`fields_stripped` | bool | Whether any variants have been removed from the result, due to being private or hidden.
+`impls`       | [[ID](#ID)] | All impls (both trait and inherent) for this type. All of the corresponding Items have `kind = "impl"`
+
+### `kind == "variant"`
+
+`inner` can be one of the 3 following objects:
+- `"plain"` (e.g. `Enum::Variant`)
+- `{"tuple": [Type]}` (e.g. `Enum::Variant(u32, String)`)
+- `{"struct": Object}` (e.g. `Enum::Variant{foo: u32, bar: String}`) in which case the `Object`
+  has a single key `"struct"` with a value that's the same object as `inner` when `kind == "struct"`.
```

Comment:
This is confusing. Do you mean that `"struct"` and `"inner"` are keys to the same object? Why have both?

Reply:
Yeah sorry that was pretty muddled, I've simplified it to not share all the fields with the struct object.